perun-ldapc/src/main/java/cz/metacentrum/perun/ldapc/processor/impl/FacilityAttributeProcessor.java

```diff
@@ -86,4 +91,24 @@ public void processAllAttributesRemoved(String msg, MessageBeans beans) {
 		}
 	}
 
-}
+	public void processVirtualAttributeChanged(String msg, MessageBeans beans) {
+		if (beans.getUser() == null || beans.getAttribute() == null || beans.getResource() == null) {
```

Comment:
I'm not getting it. Why those beans are checked ? If this is a change about virtual facility attribute, we should check presence of Attribute and Facility within the beans read from message.